src/main/java/org/damap/base/conversion/AbstractTemplateExportScienceEuropeComponents.java

```diff
@@ -320,6 +321,41 @@ public void datasetsInformation() {
     else addReplacement(replacements, "[targetaudience]", "");
   }
 
+  /** storageIntroInformation */
+  public void storageIntroInformation() {
+    String coordinatorFullName;
+
+    if (projectCoordinator != null) {
+      coordinatorFullName =
+          projectCoordinator.getFirstName() + " " + projectCoordinator.getLastName();
+    } else {
+      coordinatorFullName = "Coordinator";
+    }
+
+    boolean usesExternalStorage =
+        dmp.getHostList().stream().anyMatch(ExternalStorage.class::isInstance);
+    boolean usesInternalStorage = dmp.getHostList().stream().anyMatch(Storage.class::isInstance);
+
+    String storageIntroReplacement = "";
+
+    if (usesExternalStorage && !usesInternalStorage) {
+      storageIntroReplacement =
+          loadResourceService.loadVariableFromResource(prop, "storageIntro.external");
+      storageIntroReplacement = storageIntroReplacement.replace("[name]", coordinatorFullName);
+
+    } else if (usesInternalStorage && !usesExternalStorage) {
+      storageIntroReplacement =
+          loadResourceService.loadVariableFromResource(prop, "storageIntro.internal");
+      storageIntroReplacement = storageIntroReplacement.replace("[name]", coordinatorFullName);
+    } else if (usesInternalStorage && usesExternalStorage) {
+      storageIntroReplacement =
+          loadResourceService.loadVariableFromResource(prop, "storageIntro.both");
+      storageIntroReplacement = storageIntroReplacement.replace("[name]", coordinatorFullName);
+    }
```

Comment:
If no storage is selected, the paragraph will be empty. Probably makes sense to add a default text for this case as well.
Each branch basically does the same: load a prop and replace the variables. Just a suggestion to reduce duplicate logic. If a placeholder is not present, nothing happens so that should be fine as well.

```suggestion
    String propName = "storageIntro.none"; 
    if (usesExternalStorage && !usesInternalStorage) {
      propName = "storageIntro.external";
    } else if (usesInternalStorage && !usesExternalStorage) {
       propName = "storageIntro.internal";
    } else if (usesInternalStorage && usesExternalStorage) {
      propName = "storageIntro.both";
    }
    String storageIntroReplacement = loadResourceService.loadVariableFromResource(prop, propName);
    storageIntroReplacement = storageIntroReplacement.replace("[name]", coordinatorFullName);
    
```

Reply:
Good idea, makes the code cleaner looking!